rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -30,6 +33,65 @@
 logger = logging.getLogger(__name__)
 
 
+def item2index_mapping(unique_items: Iterable[Text]) -> Dict[Text, int]:
+    """Computes a mapping that maps each given label to a unique int index.
+
+    The computed mapping is consistent across sessions since the unique_items are
+    sorted before the mappin is computed.
+
+    Raises:
+      ValueError if the given labels are not unique
+    """
+    if len(set(unique_items)) != len(unique_items):
+        raise ValueError("The given values are not unique.")
+    # sorted to assure we get the same order with each session (just in case this
+    # list was generated via from some set/dict/...)
+    return {
+        feature_state: idx for idx, feature_state in enumerate(sorted(unique_items))
+    }
+
+
+def get_name_attribute(sub_state: SubState) -> Optional[Text]:
+    """ TODO: this could go into some SubState class....
+    """
+    # there is always either INTENT or ACTION_NAME
+    name_attributes = [
+        attribute
+        for attribute in sub_state.keys()
+        if attribute in {INTENT, ACTION_NAME}
+    ]
+    if len(name_attributes) == 0:
+        raise ValueError("Expected INTENT or ACTION_NAME.")
+    elif len(name_attributes) > 1:  # TODO: added this because the comment said so..
+        raise ValueError("Expected either INTENT or ACTION_NAME, not both.")
+    return name_attributes[0]
+
+
+def convert_sparse_sequence_to_sentence_features(
+    features: List["Features"],
+) -> List["Features"]:
+    """Grabs all sparse sequence features and turns them into sparse sentence features.
+
+    That is, this function filters all sparse sequence features and then
+    turns each of these sparse sequence features into sparse sentence feature
+    by summing over their first axis, respectively.
```

Comment:
This doesn't always work. For example, the "natural" operation to combine semantic map embeddings is not a sum.

Reply:
Since this is something that depends on the kind of Features and their origin (which means this will not make separate state featurizer classes necessary), I think we'll leave this in here as TODO (for a future Feature rework or so?)